cpp/src/arrow/type_traits.h

```diff
@@ -389,193 +389,258 @@ using void_t = typename make_void<Ts...>::type;
 // Useful type predicates
 //
 
-template <typename T>
-using is_number_type = std::is_base_of<NumberType, T>;
+// only in C++14
```

Comment:
Is there a conflict when compiling with `-std=c++14`?

Reply:
There shouldn't because it's in arrow namespace.